configmap.bzl

```diff
@@ -0,0 +1,63 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_k8s//k8s:object.bzl", "k8s_object")
+
+def _impl(ctx):
+  args = [
+    "--output=%s" % ctx.outputs.output.path,
+    "--name=%s" % ctx.label.name[:-len(".generated-yaml")],
+    "--namespace=%s" % ctx.attr.namespace,
+  ]
+  for key, value in ctx.attr.labels.items():
+    args.append("--label=%s=%s" % (key, value))
+  for target, name in ctx.attr.data.items():
+    f = target.files.to_list()[0]
+    args.append(ctx.expand_location("--data=%s=%s" % (name, f.path)))
```

Comment:
It is a little confusing that you switch the `target->name` mapping to `name->target` here. Either ordering is fine, but it is probably best to be consistent if possible.